compiler/qsc_passes/src/capabilitiesck.rs

```diff
@@ -0,0 +1,538 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#[cfg(test)]
+mod base;
+
+#[cfg(test)]
+mod adaptive;
+
+#[cfg(test)]
+mod adaptive_plus_integers;
+
+#[cfg(test)]
+pub mod common;
+
+use miette::Diagnostic;
+use qsc_data_structures::span::Span;
+use qsc_fir::{
+    fir::{
+        Block, BlockId, CallableImpl, Expr, ExprId, ExprKind, Global, Ident, Item, ItemKind,
+        LocalItemId, LocalVarId, Package, PackageLookup, Pat, PatId, PatKind, Res, SpecDecl,
+        SpecImpl, Stmt, StmtId, StmtKind,
+    },
+    ty::FunctorSetValue,
+    visit::Visitor,
+};
+use qsc_frontend::compile::RuntimeCapabilityFlags;
+use qsc_rca::{ComputeKind, ItemComputeProperties, PackageComputeProperties, RuntimeFeatureFlags};
+use thiserror::Error;
+
+#[derive(Clone, Debug, Diagnostic, Error)]
+pub enum Error {
+    #[error("cannot use a dynamic boolean value")]
+    #[diagnostic(help(
+        "using a dynamic boolean value, a boolean value that depends on a measurement result, is not supported by the target"
```

Comment:
I'm sure there is lots of room for tweaking how we communicate these things to users, but as a first feedback I think repeating the term "dynamic boolean" is not needed, and "target" could be more clearly stated as "current target" to imply that this is configurable. So something like this (and similar for the others below):

```suggestion
        "using a bool value that depends on a measurement result is not supported by the current target"
```